core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -189,7 +190,10 @@ protected ShardValidateQueryResponse shardOperation(ShardValidateQueryRequest re
             }
             if (request.rewrite()) {
                 explanation = getRewrittenQuery(searcher.searcher(), searchContext.query());
-            }   
+            }
+        } catch (QueryShardException e) {
+            valid = false;
+            error = e.getDetailedMessage();
         } catch (QueryParsingException e) {
```

Comment:
no, the catch there should be removed as well....that is what my IDE tells me :)